app/src/main-process/main.ts

```diff
@@ -685,6 +694,15 @@ app.on('ready', () => {
 
   ipcMain.handle('save-guid', (_, guid) => saveGUIDFile(guid))
 
+  ipcMain.handle(
+    'get-title-bar-style',
+    async () => readTitleBarConfigFileSync().titleBarStyle
+  )
+
+  ipcMain.handle('save-title-bar-style', (_, titleBarStyle) =>
+    saveTitleBarConfigFile({ titleBarStyle })
+  )
```

Comment:
:art: we should await the promise that this function returns

```suggestion
  ipcMain.handle(
    'save-title-bar-style',
    async (_, titleBarStyle) => await saveTitleBarConfigFile({ titleBarStyle })
  )
```

Reply:
That might be redundant.

```js
const someAsyncFunction = async () => "Resolved value"

const current = () => someAsyncFunction()
const suggested = async () => await someAsyncFunction()

current() // Promise {<fulfilled>: 'Resolved value'}
await current() // 'Resolved value'

suggested() // Promise {<fulfilled>: 'Resolved value'}
await suggested() // 'Resolved value'
```

Reply:
Try using a function that actually returns a promise when it's invoked:

```ts
const someAsyncFunction = () => Promise.resolve("Resolved value")
```

Or perhaps a promise that does some work before returning a value.

The `writeFile` function you're using within `saveTitleBarConfigFile` returns a `Promise`, and without returning or `await`-ing the promise you could create a dangling promise.